rust/frameworks/tw_utxo/src/transaction/standard_transaction/mod.rs

```diff
@@ -0,0 +1,356 @@
+const LOCKTIME_SIZE: usize = 4;
+const WITNESS_FLAG_MARKER: usize = 2;
+const OUT_POINT_SIZE: usize = 36;
+const SEQUENCE_SIZE: usize = 4;
+const VALUE_SIZE: usize = 8;
+
+// The Segwit scale factor (witnesses are deducted).
+const SEGWIT_SCALE_FACTOR: usize = 4;
+
+/// A standard Bitcoin transaction.
+///
+/// # Important
+///
+/// Do not add chain specific fields to the standard Bitcoin [`Transaction`].
+/// Otherwise, consider adding a new type of transaction that implements the [`UnsignedTransaction`] trait.
+#[derive(Clone, Debug)]
+pub struct Transaction {
+    /// Transaction data format version (note, this is signed).
+    pub version: i32,
+    /// Unsigned transaction inputs.
+    pub inputs: Vec<TransactionInput>,
+    /// Transaction outputs.
+    pub outputs: Vec<TransactionOutput>,
+    /// The block number or timestamp at which this transaction is unlocked.
+    ///
+    /// | Value          | Description
+    /// |----------------|------------
+    /// |  0             | Not locked
+    /// | < 500000000    | Block number at which this transaction is unlocked
+    /// | >= 500000000   | UNIX timestamp at which this transaction is unlocked
+    ///
+    /// If all inputs have final (`0xffffffff`) sequence numbers then `lockTime` is irrelevant.
+    /// Otherwise, the transaction may not be added to a block until after `lockTime`.
+    pub locktime: u32,
+}
+
+impl TransactionInterface for Transaction {
+    type Input = TransactionInput;
+    type Output = TransactionOutput;
+
+    fn version(&self) -> i32 {
+        self.version
+    }
+
+    fn inputs(&self) -> &[Self::Input] {
+        &self.inputs
+    }
+
+    fn inputs_mut(&mut self) -> &mut [Self::Input] {
+        &mut self.inputs
+    }
+
+    fn replace_inputs(&mut self, inputs: Vec<Self::Input>) {
+        self.inputs = inputs;
+    }
+
+    fn outputs(&self) -> &[Self::Output] {
+        &self.outputs
+    }
+
+    fn outputs_mut(&mut self) -> &mut [Self::Output] {
+        &mut self.outputs
+    }
+
+    fn change_amount(&self) -> Option<Amount> {
+        Some(self.outputs.last()?.value)
+    }
+
+    fn set_change_amount(&mut self, change: Amount) -> bool {
+        // TODO: Is it a good idea to just assume that the last output is the
+        // change output?
+        let Some(change_output) = self.outputs.last_mut() else {
+            return false;
+        };
+
+        change_output.value = change;
+
+        true
+    }
+
+    fn replace_outputs(&mut self, outputs: Vec<Self::Output>) {
+        self.outputs = outputs;
+    }
+
+    fn has_witness(&self) -> bool {
+        self.inputs.iter().any(|input| input.has_witness())
+    }
+
+    fn locktime(&self) -> u32 {
+        self.locktime
+    }
+}
+
+impl Transaction {
+    pub fn txid(&self) -> Vec<u8> {
+        // TODO
+        vec![]
+    }
+    /// Returns the same transaction with [`TransactionInput::script_witness`] being empty.
+    /// TODO: Why?
+    pub fn without_witness(&self) -> Transaction {
```

Comment:
That's needed to calculate a transaction ID.
https://github.com/trustwallet/wallet-core/blob/13cdf838f334a1101774c849ab82983b2e319701/src/Bitcoin/Signer.cpp#L74-L77
So basically, this should be used in `Transaction::txid`